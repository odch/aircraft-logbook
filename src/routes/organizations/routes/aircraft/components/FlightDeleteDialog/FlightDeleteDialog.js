import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage } from 'react-intl'
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import DeleteButton from '../../../../../../components/DeleteButton'
import {
  aircraft as aircraftShape,
  flight as flightShape,
  intl as intlShape
} from '../../../../../../shapes'
import { formatDate, formatTime } from '../../../../../../util/dates'

const FlightAttribute = props => (
  <DialogContentText className={props.className}>
    <FormattedMessage id={`flight.delete.dialog.${props.label}`} />
    :&nbsp;
    {props.value}
  </DialogContentText>
)

FlightAttribute.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  className: PropTypes.string
}

const styles = {
  attributesWrapper: {
    marginTop: '1em'
  },
  attribute: {
    marginBottom: 1
  }
}

class FlightDeleteDialog extends React.Component {
  handleClose = () => {
    if (!this.props.submitted && this.props.onClose) {
      this.props.onClose()
    }
  }

  render() {
    const { flight, submitted, classes, onConfirm } = this.props
    return (
      <Dialog onClose={this.handleClose} data-cy="flight-delete-dialog" open>
        <DialogTitle>
          <FormattedMessage id="flight.delete.dialog.title" />
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            <FormattedMessage id="flight.delete.dialog.text" />
          </DialogContentText>
          <div className={classes.attributesWrapper}>
            <FlightAttribute
              label="date"
              value={formatDate(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              )}
              className={classes.attribute}
            />
            <FlightAttribute
              label="pilot"
              value={`${flight.pilot.firstname} ${flight.pilot.lastname}`}
              className={classes.attribute}
            />
            <FlightAttribute
              label="departureaerodrome"
              value={flight.departureAerodrome.name}
              className={classes.attribute}
            />
            <FlightAttribute
              label="destinationaerodrome"
              value={flight.destinationAerodrome.name}
              className={classes.attribute}
            />
            <FlightAttribute
              label="blockofftime"
              value={formatTime(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              )}
              className={classes.attribute}
            />
            <FlightAttribute
              label="blockontime"
              value={formatTime(
                flight.blockOnTime,
                flight.destinationAerodrome.timezone
              )}
              className={classes.attribute}
            />
          </div>
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleClose} disabled={submitted}>
            <FormattedMessage id="flight.delete.dialog.buttons.cancel" />
          </Button>
          <DeleteButton
            onClick={onConfirm}
            color="secondary"
            variant="contained"
            label={this.props.intl.formatMessage({
              id: 'flight.delete.dialog.buttons.delete'
            })}
            data-cy="flight-delete-button"
            disabled={submitted}
            inProgress={submitted}
          />
        </DialogActions>
      </Dialog>
    )
  }
}

FlightDeleteDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  aircraft: aircraftShape.isRequired,
  flight: flightShape.isRequired,
  submitted: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  onConfirm: PropTypes.func,
  onClose: PropTypes.func,
  intl: intlShape
}

export default withStyles(styles)(injectIntl(FlightDeleteDialog))
