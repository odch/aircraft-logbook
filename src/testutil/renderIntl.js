import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = component => renderer.create(renderInIntlProvider(component))

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
