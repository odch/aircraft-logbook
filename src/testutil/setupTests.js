import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
