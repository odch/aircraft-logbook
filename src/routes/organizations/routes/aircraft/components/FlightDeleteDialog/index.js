import FlightDeleteDialog from './FlightDeleteDialog'

export default FlightDeleteDialog
