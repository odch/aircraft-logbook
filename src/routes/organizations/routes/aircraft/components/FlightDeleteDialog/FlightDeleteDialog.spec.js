import React from 'react'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import FlightDeleteDialog from './FlightDeleteDialog'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('FlightDeleteDialog', () => {
            it('renders correctly', () => {
              const aircraft = {
                id: 'o7flC7jw8jmkOfWo8oyA',
                registration: 'HBKLA'
              }
              const flight = {
                id: 'sStfyLd2XArT7oUZPFDn',
                departureAerodrome: {
                  name: 'Lommis',
                  timezone: 'Europe/Zurich'
                },
                destinationAerodrome: {
                  name: 'Lommis',
                  timezone: 'Europe/Zurich'
                },
                blockOffTime: {
                  toDate: () => Date.parse('2018-11-20 10:00 GMT+0100')
                },
                blockOnTime: {
                  toDate: () => Date.parse('2018-11-20 11:10 GMT+0100')
                },
                takeOffTime: {
                  toDate: () => Date.parse('2018-11-20 10:10 GMT+0100')
                },
                landingTime: {
                  toDate: () => Date.parse('2018-11-20 11:00 GMT+0100')
                },
                pilot: {
                  firstname: 'Max',
                  lastname: 'Muster'
                }
              }
              const renderedValue = renderIntlMaterial(
                <FlightDeleteDialog
                  organizationId="my_org"
                  aircraft={aircraft}
                  flight={flight}
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
