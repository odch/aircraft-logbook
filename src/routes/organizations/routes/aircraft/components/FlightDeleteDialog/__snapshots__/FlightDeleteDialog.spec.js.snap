// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components FlightDeleteDialog renders correctly 1`] = `
"<div class=\\"MuiModal-root-14 MuiDialog-root-2 MuiDialog-scrollPaper-3\\" role=\\"dialog\\" data-cy=\\"flight-delete-dialog\\">
  <div class=\\"MuiBackdrop-root-16\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div class=\\"MuiPaper-root-18 MuiPaper-elevation24-44 MuiPaper-rounded-19 MuiDialog-paper-5 MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiDialogTitle-root-45\\">
      <h2 class=\\"MuiTypography-root-46 MuiTypography-title-52\\"><span>Flug löschen</span></h2>
    </div>
    <div class=\\"MuiDialogContent-root-72\\">
      <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Bist du sicher, dass du diesen Flug löschen möchtest?</span></p>
      <div class=\\"InjectIntl-FlightDeleteDialog--attributesWrapper-1\\">
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Datum</span>:&nbsp;20.11.2018</p>
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Pilot</span>:&nbsp;Max Muster</p>
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Startflugplatz</span>:&nbsp;Lommis</p>
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Zielflugplatz</span>:&nbsp;Lommis</p>
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Block off</span>:&nbsp;10:00</p>
        <p class=\\"MuiTypography-root-46 MuiDialogContentText-root-73 MuiTypography-subheading-53 MuiTypography-colorTextSecondary-70\\"><span>Block on</span>:&nbsp;11:10</p>
      </div>
    </div>
    <div class=\\"MuiDialogActions-root-74\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-102 MuiButton-root-76 MuiButton-text-78 MuiButton-flat-81 MuiDialogActions-action-75\\" type=\\"button\\"><span class=\\"MuiButton-label-77\\"><span>Abbrechen</span></span><span class=\\"MuiTouchRipple-root-105\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-102 MuiButton-root-76 MuiButton-contained-87 MuiButton-containedSecondary-89 MuiButton-raised-90 MuiButton-raisedSecondary-92 DeleteButton-button-112\\" type=\\"button\\" data-cy=\\"flight-delete-button\\"><span class=\\"MuiButton-label-77\\"><svg class=\\"MuiSvgIcon-root-114 DeleteButton-leftIcon-113\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg>Flug löschen</span><span class=\\"MuiTouchRipple-root-105\\"></span></button></div>
  </div>
</div>"
`;
